### PR TITLE
Avoid unnessary string copies in istringstream

### PR DESCRIPTION
**Pull request by lhofhansl**
See #13559

This PR is WIP

## How to test

Load any world. Make sure everything works.
I observed a 5-8% speed up in load time (especially for large viewing_ranges).

I wondering whether this helps with #14015, since it potentially can prevent a LOT of small heap allocations.

### CODE REVIEW

**Comment by lhofhansl** [comment]
Hmm... Does not compile on Android or MacOS. I'll look into that. Compiles fine on Linux GCC 13.2.

**Comment by lhofhansl** [comment]
And there are some issues when passing the zcistream to deflate. Maybe I shot too quickly.

**Comment by lhofhansl** [comment]
Ok. This fixes the readLongString bit. Not quite as pretty, but it completely avoid copying long string out of the network packet.

**Comment by lhofhansl** [comment]
Meh... Calling the istream constructor (instead of just istream::init) actually makes it slower.
I give up, I think. String streams are just slow.

File: src/network/networkpacket.h [hunk 1]
```diff
@@ -63,6 +63,7 @@ class NetworkPacket
 	NetworkPacket &operator<<(const std::wstring &src);
 
 	std::string readLongString();
+	const char *readLongString(u32 strLen);
```
Comment (sfan5) [inline src/network/networkpacket.h:66]: FWIW we're c++17 so `std::string_view` is what you want here

**Comment by lhofhansl** [comment]
FYI. I ran Minetest through heaptrack and this did not appear to make huge dent. I'll give up on this line of thinking.

**Pull request closed without merging**